PMax - Add Tooltip for SSC Campaigns

jorgemd24 opened the pull request:
### Changes proposed in this Pull Request:

This PR adds the tooltip in the GLA Reports Section when hovering over the old campaigns once the campaign has been converted.  See more context in the following links:

- [Design Proposal i2: Performance Max Migration (Milestone 2)](https://woodesigngrowcontractor.wordpress.com/2022/04/20/design-proposal-i2-performance-max-migration-milestone-2/)
- [Figma File](https://href.li/?https://www.figma.com/file/eQ9O4m2flzrcAiDMXkOW0m/Google-Listings-%26-Ads-v1.x?node-id=7840%3A127644)

Design Example: 
![ezgif com-gif-maker](https://user-images.githubusercontent.com/2488994/167656814-475b69ed-0fed-4223-b3d5-6c4d1521aafb.gif)

### Detailed test instructions:

#### In case your account has old SSC Campaigns and PMax campaigns in the Reports - Program section:

1. In wp_options check if exists following option, `gla_campaign_convert_status`, otherwise refresh the page.
2. In the option_value add `a:2:{s:6:"status";s:9:"converted";s:7:"updated";i:1652254871;}`
3.  Go to the Reports Section -> Programs
4. Hover the campaign name of one of the SSC Campaigns and the text: "This campaign has been upgraded to Performance Max" should be displayed.

#### In case your account does not have reporting data:
1.  Add the following snippet in your `google-listings-and-ads.php`

```
add_filter(
	'woocommerce_gla_prepared_response_ads-reports-programs',
	function( $response ) {
		return json_decode( file_get_contents( __DIR__ . '/tests/mocks/ads/reports/programs.json' ), true ) ?: [];
	},
);
```
3. Go to the Reports Section -> Programs
4. The first campaign is an old SSC campaign, so hover over the name and the tooltip should be displayed.

<!--
Optional.
Enter a summary of all changes in this Pull Request, which will be added to the changelog if accepted.
Each line should start with `(Fix|Add|Tweak|Update) - `, for example:
> Fix - I took care of something that wasn't working.
> Add - I added something new that's pretty cool.
> Tweak - I made a small change.
> Update - I made big changes to something that wasn't broken.

Leave the "Changelog entry" header in place completely empty, without any summary if no changelog entry is needed.
If you remove the "Changelog entry" header, the title of Pull Request will be used as the changelog entry.
-->

### Additional details
The PHP unit tests are failing because of WC 6.5 (released 10/05/22) is not including feature-config.php. I have tested locally with WC 6.4 and it is passing all tests.

```
PHP Warning:  require_once(/tmp/wordpress/src/wp-content/plugins/woocommerce//includes/react-admin/feature-config.php): failed to open stream: No such file or directory in /tmp/wordpress/src/wp-content/plugins/woocommerce/src/Internal/Admin/FeaturePlugin.php on line 81
```

### Changelog entry

> Add - Tooltip in reports section for SSC Campaigns

jorgemd24 committed:
Add constant campaign types

jorgemd24 committed:
Add function isCampaignConverted

jorgemd24 committed:
Add test is campaign converted

jorgemd24 committed:
Add Name Campaign Cell component

jorgemd24 committed:
Add tests NameCampaignCell component

jorgemd24 committed:
Update CellName in reports table

jorgemd24 committed:
Add campaign type in reports controller

jorgemd24 committed:
Check if campaign type is set

jorgemd24 committed:
Update test to check the campaign type

jorgemd24 committed:
Update reports unit test

jorgemd24 committed:
Add JSDoc

jorgemd24 committed:
Add css campaign name cell

jorgemd24 committed:
Update schema reports controller

jorgemd24 committed:
Rename component

jorgemd24 committed:
Update mock reports programs

jorgemd24 committed:
Refactor isConverted

jorgemd24 committed:
Rename files

puntope commented on js/src/reports/programs/campaign-name-cell.js:
```diff
@@ -0,0 +1,42 @@
+ *
+ * @param {Object} props React props.
+ * @param {string} props.name Campaign Name
+ * @param {string} props.isConverted Weather the campaign is converted
```
This is boolean type I guess

puntope commented on js/src/reports/programs/campaign-name-cell.js:
```diff
@@ -0,0 +1,42 @@
+ * @param {Object} props React props.
+ * @param {string} props.name Campaign Name
+ * @param {string} props.isConverted Weather the campaign is converted
+ * @return {JSX.Element} {@link Notice} element with the info message and the link to the documentation.
```
❓ 
What is this `{@link Notice} ` for? 🤔

jorgemd24 replied:
Yeah `{@link Notice}` should not be there, it was there because I was working with the Notice component in another PR. Thanks for catching this. I have updated: [03e26db](https://github.com/woocommerce/google-listings-and-ads/pull/1496/commits/03e26db7b4228c7bf895f85105e5662830397777)

puntope commented on js/src/reports/programs/campaign-name-cell.scss:
```diff
@@ -0,0 +1,5 @@
+.gla-reports__tooltip-campaing-name {
+	.app-tooltip__children-container {
+		width: 100%;
+	}
+}
```
💅 / ❓ 

There is a typo in the class name --> `campaing` instead of `campaign`

Besides that, I removed the wrap with selector and works the same. What is exactly the intent of the styled wrapper?

jorgemd24 replied:
I fixed the typo: [f2ad4aa](https://github.com/woocommerce/google-listings-and-ads/pull/1496/commits/f2ad4aac518891559f3c66aa0ee3c3abce82dfed)

The intention is to have accessible the tooltip through the full campaign name cell. If I remove the class, the tooltip will show only if we hover over the name.

puntope reviewed (commented):
Hey @jorgemd24 thanks for implementing this. Overall lucks pretty good. I was able to see the tooltip in the converted campaign.

Left some minor comments to be discussed / addressed.

Thanks in advance.

puntope commented:
> The PHP unit tests are failing because of WC 6.5 (released 10/05/22) is not including feature-config.php. I have tested locally with WC 6.4 and it is passing all tests.

Thanks for the info :) . I guess[ the fix is already in merged in develop](https://github.com/woocommerce/google-listings-and-ads/pull/1498) so probably is fixable by merging/rebasing develop in `feature/pmax-milestone-2`

jorgemd24 committed:
Update JSDoc

jorgemd24 committed:
Fix typo

jorgemd24 committed:
Update JSDoc return element

jorgemd24 commented:
Thanks for your comments @puntope! 

I have addressed your comments and I left a comment about the CSS.

>I guesshttps://github.com/woocommerce/google-listings-and-ads/pull/1498 so probably is fixable by merging/rebasing develop in feature/pmax-milestone

Yeah, I will merge develop into the base branch, thanks!

puntope reviewed (approved):
APPROVED ✅ 

Thanks for the adjustments. 

I tested using the filter hook with the json mocked data and I saw the tooltip in the name cell.

However, looking at the UX proposal I'm wondering if we should show the tooltip in the row instead of in the cell. 

> _When hovering over the old campaigns, a tooltip shows up that says “This campaign has been upgraded to Performance Max“. We tried greying out the old campaigns but they look disabled/unclickable. So we went with keeping them all in the same color._ 

I approved this in advance but it's worth it to double check with design if they wanted the tooltip all over the row or just in the name which can be adjusted in future PR if needed. 

Thanks
